cmake/third_party/aws_sdk.cmake

```diff
@@ -1,44 +1,73 @@
 include(ExternalProject)
 include(GNUInstallDirs)
+include(${CMAKE_SOURCE_DIR}/cmake/helpers.cmake)
 
-# Don't add external project if extension is not enabled.
-if(NOT ENABLE_S3)
-    return()
+config_choice(
+    IMPORT_S3_SDK
+    "Specify how to import the S3 SDK"
+    OPTIONS
+        "none;IMPORT_S3_SDK_NONE;NOT ENABLE_S3"
+        "package;IMPORT_S3_SDK_PACKAGE;ENABLE_S3"
+        "external;IMPORT_S3_SDK_EXTERNAL;ENABLE_S3"
+)
+ # Skip the AWS SDK build step if the extension is not enabled.
+ if(NOT ENABLE_S3)
+     return()
+ endif()
+ 
+if(IMPORT_S3_SDK_NONE)
+    message(FATAL_ERROR "Cannot enable S3 extension without specifying an IMPORT_S3_SDK method (package, none).")
```

Comment:
Oh whoops. I realize this might've been from me. It should be `(package, external)`